modules/oip042/api.go

```diff
@@ -48,3 +56,144 @@ func handleLatest(w http.ResponseWriter, r *http.Request) {
 	)
 	httpapi.RespondSearch(w, searchService)
 }
+
+/**
+This method will return the record requested with all edits applied. So, the most recent version of the record.
+*/
+func handleGetLatestEdit(response http.ResponseWriter, request *http.Request) {
+	vars := mux.Vars(request)
+	originalTxid := vars["originalTxid"]
+	var opts = mux.Vars(request)
+
+	query := elastic.NewBoolQuery().Must(
+		elastic.NewTermQuery("meta.blacklist.blacklisted", false),
+		elastic.NewTermQuery("meta.deactivated", false),
+		elastic.NewTermQuery("meta.originalTxid", originalTxid),
+		elastic.NewTermQuery("meta.latest", true),
+	)
+
+	if n, ok := opts["nsfw"]; ok {
+		nsfw, _ := strconv.ParseBool(n)
+		if !nsfw {
+			query.MustNot(elastic.NewTermQuery("artifact.info.nsfw", true))
+		}
+		log.Info("nsfw: %t", nsfw)
+	}
+
+	searchService := httpapi.BuildCommonSearchService(
+		request.Context(),
+		[]string{oip042ArtifactIndex},
+		query,
+		[]elastic.SortInfo{
+			{Field: "meta.time", Ascending: false},
+			{Field: "meta.txid", Ascending: true},
+		},
+		o42ArtifactFsc,
+	)
+	httpapi.RespondSearch(response, searchService)
+}
+
+/**
+This method will return the record requested with the version requested. The version will be specified using the transaction ID.
+*/
+func handleGetForVersion(response http.ResponseWriter, request *http.Request) {
+	vars := mux.Vars(request)
+	originalTxid := vars["originalTxid"]
+	editRecordTxid := vars["editRecordTxid"]
+	var opts = mux.Vars(request)
+
+	var query *elastic.BoolQuery
+
+	if originalTxid == editRecordTxid {
+		query = elastic.NewBoolQuery().Must(
+			elastic.NewTermQuery("meta.blacklist.blacklisted", false),
+			elastic.NewTermQuery("meta.deactivated", false),
+			elastic.NewTermQuery("meta.originalTxid", originalTxid),
```

Comment:
You are absolutely correct, I will go ahead and implement this change :)

Reply:
Actually, since this API call is grabbing a specific edit, so it doesn't need it's txid swapped.

The `originalTxid` variable holds the initial txid of the first Record published, not of the edit :)